crates/rpc/rpc/src/eth/sim_bundle.rs

```diff
@@ -1,15 +1,66 @@
 //! `Eth` Sim bundle implementation and helpers.
 
-use std::sync::Arc;
-
-use alloy_rpc_types_mev::{SendBundleRequest, SimBundleOverrides, SimBundleResponse};
+use alloy_eips::BlockNumberOrTag;
+use alloy_primitives::U256;
+use alloy_rpc_types::{calc_excess_blob_gas, BlockId};
+use alloy_rpc_types_mev::{
+    BundleItem, Inclusion, Privacy, RefundConfig, SendBundleRequest, SimBundleLogs,
+    SimBundleOverrides, SimBundleResponse, Validity,
+};
 use jsonrpsee::core::RpcResult;
+use reth_chainspec::EthChainSpec;
+use reth_evm::{ConfigureEvm, ConfigureEvmEnv};
+use reth_primitives::{
+    revm_primitives::db::{DatabaseCommit, DatabaseRef},
+    TransactionSigned,
+};
+use reth_provider::{ChainSpecProvider, HeaderProvider};
+use reth_revm::database::StateProviderDatabase;
 use reth_rpc_api::MevSimApiServer;
-use reth_rpc_eth_api::helpers::{Call, EthTransactions, LoadPendingBlock};
-use reth_rpc_eth_types::EthApiError;
+use reth_rpc_eth_api::{
+    helpers::{Call, EthTransactions, LoadPendingBlock},
+    FromEthApiError, RpcNodeCore,
+};
+use reth_rpc_eth_types::{utils::recover_raw_transaction, EthApiError};
 use reth_tasks::pool::BlockingTaskGuard;
+use revm::{
+    db::CacheDB,
+    primitives::{Address, EnvWithHandlerCfg, ResultAndState, SpecId, TxEnv},
+};
+use std::{sync::Arc, time::Duration};
 use tracing::info;
 
+/// A flattened representation of a bundle item containing transaction and associated metadata.
+#[derive(Clone, Debug)]
+pub struct FlattenedBundleItem {
+    /// The signed transaction
+    pub tx: TransactionSigned,
+    /// The address that signed the transaction
+    pub signer: Address,
+    /// Whether the transaction is allowed to revert
+    pub can_revert: bool,
+    /// Item-level inclusion constraints
+    pub inclusion: Inclusion,
+    /// Optional validity constraints for the bundle item
+    pub validity: Option<Validity>,
+    /// Optional privacy settings for the bundle item
+    pub privacy: Option<Privacy>,
+}
+
+/// Maximum bundle depth
+const MAX_NESTED_BUNDLE_DEPTH: usize = 5;
+
+/// Maximum body size
+const MAX_BUNDLE_BODY_SIZE: usize = 50;
+/// Default simulation timeout
+const DEFAULT_SIM_TIMEOUT: Duration = Duration::from_secs(5);
+
+/// Maximum simulation timeout
+const MAX_SIM_TIMEOUT: Duration = Duration::from_secs(30);
+
+/// Maximum payout cost
+const SBUNDLE_PAYOUT_MAX_COST: u64 = 30_000;
```

Comment:
const should go top of file

Reply:
resolved in [cf2072f](https://github.com/paradigmxyz/reth/pull/12218/commits/cf2072f669c0ffe8a345d57bc4ae3d804ce02acf)